woocommerce/woocommerce-gutenberg-products-block
# Fix: Spilled icecream error with Filter Products by Attribute when no attribute was selected

This PR:
- Fixes the Spilled ice cream error on the front end when the site has no attribute or no attribute was selected for the block.
- Adds a warning in the editor when users didn't select any attribute.
- Adds a warning in the editor when the site doesn't have any attribute term assigned to products.
- Prevents flashing title on the front-end if the users didn't select attribute or the site doesn't have any term assigned.
- Fixes the preview error when previewing the block without selecting any attributed, see https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/2441#issuecomment-927318176.

Fixes #2441




### Screenshots
New warning messages:

![image](https://user-images.githubusercontent.com/5423135/134812948-e87d54e2-26d5-43ef-b28c-a52c1a9f3e18.png)



### Testing

#### Automated Tests
### Manual Testing

How to test the changes in this Pull Request:

1. Adding a new Filter by Attribute block.
2. Click `Save` button without selecting any attribute.
3. See the warning message.
4. Edit the block, select an attribute that isn't assigned to any product.
5. See the warning message.



### Performance Impact



### Changelog

> Fix: Render error of Filter by Attribute block when no attribute was selected.
